linera-base/src/crypto/secp256k1.rs

```diff
@@ -119,30 +163,29 @@ impl FromStr for Secp256k1PublicKey {
     type Err = CryptoError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let pk = secp256k1::PublicKey::from_str(s).map_err(CryptoError::Secp256k1Error)?;
-        Ok(Secp256k1PublicKey(pk))
+        hex::decode(s)?.as_slice().try_into()
     }
 }
 
 impl TryFrom<&[u8]> for Secp256k1PublicKey {
     type Error = CryptoError;
 
     fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
-        let pk = secp256k1::PublicKey::from_slice(value).map_err(CryptoError::Secp256k1Error)?;
-        Ok(Secp256k1PublicKey(pk))
+        Self::from_bytes(value)
     }
 }
 
 impl fmt::Display for Secp256k1PublicKey {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let s = hex::encode(self.0.serialize());
-        write!(f, "{}", s)
+        let str = hex::encode(self.0.to_encoded_point(true).to_bytes());
+        write!(f, "{}", str)
     }
 }
 
 impl fmt::Debug for Secp256k1PublicKey {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{}", hex::encode(&self.0.serialize()[0..9]))
+        let str = hex::encode(&self.0.to_encoded_point(true).to_bytes()[0..9]);
+        write!(f, "{}..", str)
```

Comment:
…then you could simplify this:

```suggestion
        write!(f, "{:9}..", self)
```

(Both comments also apply to signatures below.)

Reply:
Done in [Apply review suggestions.](https://github.com/linera-io/linera-protocol/pull/3407/commits/8f530c26831f602c5a89abc74abc7445a02389d6).